### PR TITLE
Matriculation management and changes to user management

### PR DESCRIPTION
**Pull request by bastihav**
# Description

Implements matriculation management api + changes to user management api

Implements matriculation user stories (still with mock data, as API is not yet deployed nor tested)

@theovier we would like to merge this branch asap, even if the matriculation management is untested, as the user management API has changed and Lagom is still waiting to deploy 0.6.0. We can test the matriculation when HL deploys.

This branch is already pretty old and will only introduce more merging problems, while it is not even actively worked on anymore.

## Type of change (remove all that don't apply)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - user management API changed, might break features. We will test (on this branch) when this PR is accepted and lagom has deployed.

# How Has This Been Tested?

We have not tested anything yet.

# Checklist: (remove all that don't apply)

- [ ] I added corresponding E2E tests (especially for bugfixes)
- [X] I added corresponding unit tests (if API code)
- [X] I have performed a self-review of my own code
- [X] My changes generate no new linting warnings or console warnings
- [X] I have updated the CHANGELOG.md to include any changes made in this PR (add WIP to the top, if there is none already)

### CODE REVIEW

File: src/components/modals/ImmatriculationHistoryModal.vue [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    export default {
+        components: {
+            Modal,
+            ImmatriculationHistory,
+            LoadingSpinner,
+        },
+        setup() {
+            const baseModal = ref();
+            let busy = ref(false);
+            let alreadyLoadedOnce = ref(false);
```
Comment (theovier) [inline src/components/modals/ImmatriculationHistoryModal.vue:37]: Can this be safely removed?

**Review by theovier — APPROVED** [review]

**Comment by bastihav** [comment]
all tests work (except for matriculation management, which is still untested).
we should implement #403 on this branch before merging, as the phone number is part of the user management target version of this branch.
@tobias-wiese @theovier

**Comment by theovier** [comment]
Looks good. 

### Modal
Maybe we should consider removing the "refresh" button from the history modal as it is very unlikely that the matriculation status changes while this menu is opened? If we choose to keep the buttons, we should make one of them a primary action button. Or we could consider removing them all and just keep the X in the top right corner, however, I'm unsure if we should at least keep the cancel button for UX reasons.

**Comment by tobias-wiese** [comment]
> 
> 
> Looks good.
> ### Modal
> 
> Maybe we should consider removing the "refresh" button from the history modal as it is very unlikely that the matriculation status changes while this menu is opened? If we choose to keep the buttons, we should make one of them a primary action button. Or we could consider removing them all and just keep the X in the top right corner, however, I'm unsure if we should at least keep the cancel button for UX reasons.

The refresh button is needed at this point as the data would else just be refreshed on page reload. But it could be the case, that a student awaits an update of his immatriculation. @bastihav an I discussed these use cases when removing the caching option from the modal.

**Comment by theovier** [comment]
> > Looks good.
> > ### Modal
> > Maybe we should consider removing the "refresh" button from the history modal as it is very unlikely that the matriculation status changes while this menu is opened? If we choose to keep the buttons, we should make one of them a primary action button. Or we could consider removing them all and just keep the X in the top right corner, however, I'm unsure if we should at least keep the cancel button for UX reasons.
> 
> The refresh button is needed at this point as the data would else just be refreshed on page reload. But it could be the case, that a student awaits an update of his immatriculation. @bastihav an I discussed these use cases when removing the caching option from the modal.

Even if the student awaits updates on his immatriculation status, I doubt that he would let the modal open and refresh it instead of refreshing the whole page.

**Comment by bastihav** [comment]
> > > Looks good.
> > > ### Modal
> > > Maybe we should consider removing the "refresh" button from the history modal as it is very unlikely that the matriculation status changes while this menu is opened? If we choose to keep the buttons, we should make one of them a primary action button. Or we could consider removing them all and just keep the X in the top right corner, however, I'm unsure if we should at least keep the cancel button for UX reasons.
> > 
> > 
> > The refresh button is needed at this point as the data would else just be refreshed on page reload. But it could be the case, that a student awaits an update of his immatriculation. @bastihav an I discussed these use cases when removing the caching option from the modal.
> 
> Even if the student awaits updates on his immatriculation status, I doubt that he would let the modal open and refresh it instead of refreshing the whole page.

As a user, I would expect reopening the modal would show new data, however this is not easily feasible as the modal is rendered in the DOM at page loading

**Pull request merged**